var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -156,6 +156,20 @@ def configure_args(self):
 
         return ["--with-zlib=%s" % spec['zlib'].prefix] + extra_args
 
+    def configure(self, spec, prefix):
+        # Run the default autotools package configure
+        super(Hdf5, self).configure(spec, prefix)
+
+        if '@:1.8.14' in spec:
```

Comment:
I am very curious about the idea of argument injection. I was literally just trying to figure out a way to get all of the following working:
```python
def configure_args(self):
def configure_args(self, spec):
def configure_args(self, spec, prefix):
```
You'll probably notice by now that the first line of almost every `configure_args` in Spack is `spec = self.spec`. It's very commonly needed, and I think it should be made available. But if we can get all 3 of those working, everyone can be happy.